File: src/main/java/ar/edu/unq/spring/modelo/clasesDePersonajes/Mago.java
package ar.edu.unq.spring.modelo.clasesDePersonajes;

import ar.edu.unq.spring.modelo.Personaje;
import jakarta.persistence.Entity;
import lombok.Getter;
import lombok.Setter;

@Getter @Setter
@Entity
public class Mago extends Personaje {

    private int mana = 0;

    public Mago() {}

    public Mago(String nombre) {
        this.setNombre(nombre);
    }
}

File: src/main/kotlin/ar/edu/unq/spring/modelo/clasesDePersonajes/Mago.kt
package ar.edu.unq.spring.modelo.clasesDePersonajes

import ar.edu.unq.spring.modelo.Personaje
import javax.persistence.Entity

@Entity
class Mago() : Personaje() {
    var mana: Int = 0

    constructor(nombre: String) : this() {
        this.nombre = nombre
    }


}
